Gems/Meshlets/Code/Source/Meshlets/MeshletsAssets.cpp

```diff
@@ -0,0 +1,547 @@
+            const meshopt_Meshlet& last = meshlets.back();
+            meshlet_vertices.resize(last.vertex_offset + last.vertex_count);
+            meshlet_triangles.resize(last.triangle_offset + ((last.triangle_count * 3 + 3) & ~3));
+            uint32_t meshletsAmount = (uint32_t) meshlets.size();
+
+            m_meshletsData.Descriptors = meshlets;
+            m_meshletsData.IndicesIndirection = meshlet_vertices;
+            m_meshletsData.EncodeTrianglesData(meshlet_triangles);
+            m_meshletsData.ValidateData((uint32_t)meshlet_vertices.size());
+            ////////////////////////////
+
+            // Add this in order to display meshlet separation - debug purpose only
+            static bool markTextureCoordinates = true;
+            if (markTextureCoordinates)
+            {
+                debugMarkMeshletsUVs(mesh);
+            }
+
+            AZ_Warning("Meshlets", false, "Successfully generated [%d] meshlets\n", meshletsAmount);
+            return meshletsAmount;
+        }
+
+        //----------------------------------------------------------------------
+        // Enhanced (Meshlet) Model Creation 
+        //----------------------------------------------------------------------
+        Data::Asset<RPI::BufferAsset> MeshletsModel::CreateBufferAsset(
+            const AZStd::string& bufferName,
+            const RHI::BufferViewDescriptor& bufferViewDescriptor,
+            RHI::BufferBindFlags bufferBindFlags,
+            const void* data
+        )
+        {
+            RPI::BufferAssetCreator creator;
+            creator.Begin(Uuid::CreateRandom());
+
+            RHI::BufferDescriptor bufferDescriptor;
+            bufferDescriptor.m_bindFlags = bufferBindFlags;
+            bufferDescriptor.m_byteCount = static_cast<uint64_t>(bufferViewDescriptor.m_elementSize) * static_cast<uint64_t>(bufferViewDescriptor.m_elementCount);
+
+            if (data)
+            {
+                creator.SetBuffer(data, bufferDescriptor.m_byteCount, bufferDescriptor);
+            }
+
+            creator.SetBufferViewDescriptor(bufferViewDescriptor);
+            creator.SetUseCommonPool(RPI::CommonBufferPoolType::StaticInputAssembly);
+
+            Data::Asset<RPI::BufferAsset> bufferAsset;
+
+            // The next line is the actual buffer asset creation
+            AZ_Error("Meshlets", creator.End(bufferAsset), "Error -- creating buffer [%s]", bufferName.c_str());
+
+            return bufferAsset;
+        }
+
+        bool MeshletsModel::ProcessBuffersData(float* position, uint32_t vtxNum)
+        {
+            for (uint32_t vtx = 0; vtx < vtxNum; ++vtx, position += 3)
+            {
+                Vector3 positionV3 = Vector3(position[0], position[1], position[2]);
+
+                float length = positionV3.GetLength();
+                const float maxVertexSize = 99.0f;
+                if (length < maxVertexSize)
```

Comment:
could use squared length
```suggestion
                float length = positionV3.GetLengthSq();
                constexpr float maxVertexSizeSq = 99.0f * 99.0f;
                if (length < maxVertexSizeSq)
```

Reply:
sure, enhanced in two places, although this is a 'nit' especially since this is only a sanity check that should be removed and this entire process should be done offline.